examples/nidaqmx/analog-input-every-n-samples-aio.py

```diff
@@ -101,15 +101,21 @@ def check_for_warning(response):
                 )
             )
 
-            # Wait for initial_metadata to ensure that the callback is registered before starting
-            # the task.
+            # Wait for initial_metadata and check for stream errors to ensure that the callback is
+            # registered before starting the task.
             await every_n_samples_stream.initial_metadata()
+            if every_n_samples_stream.done():
```

Comment:
If wait for initial metadata doesn't update the done flag for whatever reason, then the error is not lost forever. The stream reader task/thread will raise an exception and handle it. Both asyncio and futures ensure that an unhandled exception in the stream reader task/thread will be reported to the user.